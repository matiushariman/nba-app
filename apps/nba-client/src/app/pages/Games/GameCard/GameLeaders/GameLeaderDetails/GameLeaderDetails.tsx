import type { GameLeaderDetailsProps } from './GameLeaderDetails.types';

const GameLeaderDetails = ({
  name,
  jerseyNum,
  teamTricode,
  position,
  personId,
}: GameLeaderDetailsProps) => (
  <div className="flex items-center pt-1 pb-1">
    <div className="mr-2 w-[24px] h-[24px]">
      <img
        className="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
        src={`https://cdn.nba.com/headshots/nba/latest/260x190/${personId}.png`}
        alt={`${name} headshot`}
        loading="lazy"
      />
    </div>
    <div className="grid">
      <p className="text-sm">{name}</p>
      <div className="grid grid-cols-3 divide-x gap-2">
        <p className="text-xs">{teamTricode}</p>
        <p className="text-xs">{`#${jerseyNum}`}</p>
        <p className="text-xs">{position}</p>
      </div>
    </div>
  </div>
);

export default GameLeaderDetails;
