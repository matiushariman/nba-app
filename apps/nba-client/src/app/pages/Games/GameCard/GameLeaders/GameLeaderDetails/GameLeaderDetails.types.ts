import type { Leaders } from '@nba-app/types-nba';

export type GameLeaderDetailsProps = Pick<
  Leaders,
  'name' | 'jerseyNum' | 'teamTricode' | 'position' | 'personId'
>;
