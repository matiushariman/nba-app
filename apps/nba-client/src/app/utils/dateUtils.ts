import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
