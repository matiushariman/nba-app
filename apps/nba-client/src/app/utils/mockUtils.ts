export const shouldUseMock = process.env['NODE_ENV'] === 'development';
