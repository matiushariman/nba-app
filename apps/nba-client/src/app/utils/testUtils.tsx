import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
