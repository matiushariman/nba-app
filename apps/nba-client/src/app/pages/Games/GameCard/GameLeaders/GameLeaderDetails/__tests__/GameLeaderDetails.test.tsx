import GameLeaderDetails from '..';

import { render } from '../../../../../../utils/testUtils';

describe('pages/Games/GameCard/GameLeaders/GameLeaderDetails', () => {
  it('should render GameLeaderDetails that matches snapshot', () => {
    const { baseElement } = render(
      <GameLeaderDetails
        personId={1}
        name="Jayson Tatum"
        jerseyNum="0"
        position="SF"
        teamTricode="BOS"
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
