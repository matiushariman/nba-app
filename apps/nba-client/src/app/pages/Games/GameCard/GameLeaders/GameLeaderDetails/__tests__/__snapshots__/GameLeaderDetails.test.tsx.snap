// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Games/GameCard/GameLeaders/GameLeaderDetails should render GameLeaderDetails that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex items-center pt-1 pb-1"
    >
      <div
        class="mr-2"
      >
        <div
          class="w-[24px] h-[24px]"
        >
          <img
            alt="Jayson Tatum headshot"
            class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
            loading="lazy"
            src="https://cdn.nba.com/headshots/nba/latest/260x190/1.png"
          />
        </div>
      </div>
      <div
        class="grid"
      >
        <p
          class="text-sm"
        >
          Jayson Tatum
        </p>
        <div>
          <p
            class="text-xs text-gray-500"
          >
            BOS
             | #
            0
             | 
            SF
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
